pkg/monitor/monitor/template.go

```diff
@@ -162,15 +98,184 @@ type MonitorConfigModel struct {
+						SourceLabels: "[__meta_kubernetes_pod_annotation_prometheus_io_path]",
+						Action:       "replace",
+						TargetLabel:  "__metrics_path__",
+						Regex:        "(.+)",
+					},
+					{
+						SourceLabels: "[__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]",
+						Action:       "replace",
+						Regex:        "([^:]+)(?::\\d+)?;(\\d+)",
+						Replacement:  "$1:$2",
+						TargetLabel:  "__address__",
+					},
+					{
+						SourceLabels: "[__meta_kubernetes_namespace]",
+						Action:       "replace",
+						TargetLabel:  "kubernetes_pod_ip",
+					},
+					{
+						SourceLabels: "[__meta_kubernetes_pod_name]",
+						Action:       "replace",
+						TargetLabel:  "instance",
+					},
+					{
+						SourceLabels: "[__meta_kubernetes_pod_label_app_kubernetes_io_instance]",
+						Action:       "replace",
+						TargetLabel:  "cluster",
+					},
+				},
+			},
+		},
+	}
+	return &c
 }
 
-func renderTemplateFunc(tpl *template.Template, model interface{}) (string, error) {
-	buff := new(bytes.Buffer)
-	err := tpl.Execute(buff, model)
+func addAlertManagerUrl(pc *PrometheusConfig, model *MonitorConfigModel) {
+	pc.Alerting = &AlertingSpec{
+		AlertManagers: []AlertmanagerSpec{
+			{
+				StaticConfigs: []StaticConfig{
+					{
+						Targets: []string{
+							model.AlertmanagerURL,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func addTlsConfig(pc *PrometheusConfig, model *MonitorConfigModel) {
+	for id, config := range pc.ScrapeConfigs {
+		if config.JobName == "tidb-cluster" {
+			config.TlsConfig.CaFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+			config.TlsConfig.CertFile = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+			config.TlsConfig.KeyFile = "/var/lib/pd-client-tls/key"
+			config.Schema = "https"
+			// This is a workaround of https://github.com/tikv/tikv/issues/5340 and should
+			// be removed after TiKV fix this issue
+			config.RelabelConfigs = append(config.RelabelConfigs, RelabelConfig{
+				SourceLabels: "[__meta_kubernetes_pod_name]",
+				Action:       "drop",
+				Regex:        ".*\\-tikv\\-\\d*$",
+			})
+		}
+		pc.ScrapeConfigs[id] = config
+	}
+	// This is a workaround of https://github.com/tikv/tikv/issues/5340 and should
+	// be removed after TiKV fix this issue
+	pc.ScrapeConfigs = append(pc.ScrapeConfigs, ScrapeConfigSpec{
+		JobName:        "tidb-cluster-tikv",
+		ScrapeInterval: "15s",
+		HonorLabels:    true,
+		KubernetesSDConfigs: []KubernetesSDConfig{
+			{
+				Role: "pod",
+				Namespaces: &NamespacesConfig{
+					Names: model.ReleaseNamespaces,
+				},
+			},
+		},
+		TlsConfig: TlsConfig{
+			InsecureSkipVerify: true,
+		},
+		Schema: "http",
+		RelabelConfigs: []RelabelConfig{
+			{
+				SourceLabels: "[__meta_kubernetes_pod_label_app_kubernetes_io_instance]",
+				Action:       "keep",
+				Regex:        model.ReleaseTargetRegex,
+			},
+			{
+				SourceLabels: "[__meta_kubernetes_pod_annotation_prometheus_io_scrape]",
+				Action:       "keep",
```

Comment:
I believe there is a enum in prometheus for this

Reply:
updated.